ensure notebook-testing script fails when notebooks fail

jameslamb opened the pull request:
## Description

The notebook-testing CI job in this repo does not actually cause a loud CI failure if any errors are detected in notebooks. That's because of a `bash` mistake I made in #1407... that PR moved notebook-checking into a function, but didn't add a `set -E` to be sure errors from inside that function were appropriately trapped.

This PR fixes that:

* ensures that notebook failures actually cause CI failures
* fixes 2 typos in `nyc_taxi_years_correlation.ipynb` code
  - *(not caught in #1422 because of this CI script bug)*

Context: https://github.com/rapidsai/cuspatial/pull/1422#pullrequestreview-2212919011

## Notes for Reviewers

### How I tested this

Originally did not skip any notebooks. Saw the failures in one of the notebooks cause an actual merge-blocking CI failure.

Build link: https://github.com/rapidsai/cuspatial/actions/runs/10199784404/job/28219162698?pr=1424

### Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

jameslamb commented on ci/test_notebooks.sh:
```diff
@@ -29,7 +29,7 @@ NBTEST="$(realpath "$(dirname "$0")/utils/nbtest.sh")"
 # Add notebooks that should be skipped here
 # (space-separated list of filenames without paths)
-SKIPNBS="binary_predicates.ipynb cuproj_benchmark.ipynb"
+SKIPNBS="binary_predicates.ipynb cuproj_benchmark.ipynb ZipCodes_Stops_PiP_cuSpatial.ipynb"
```
proposing skipping this until https://github.com/rapidsai/cuspatial/issues/1426 is resolved

jakirkham replied:
Now that the aforementioned issue has been resolved, can we please readd zipcodes?

```suggestion
SKIPNBS="binary_predicates.ipynb cuproj_benchmark.ipynb"
```

jameslamb replied:
ah yes! good catch, totally agree.

jameslamb commented:
This is ready for review. The failing unit test jobs are unrelated... they're the result of #1427, which will hopefully be fixed by @mroeschke 's PR: #1428

jakirkham reviewed (commented):
Thanks James! 🙏

Looks good. Had one minor suggestion

jakirkham commented:
There were a couple CI jobs that appear to have had issues starting up (this is well before anything cuSpatial related AFAICT)

Have gone ahead and restarted those failing jobs. Hopefully that clears out the issue

jakirkham commented:
/merge

jameslamb commented:
Thanks @jakirkham !

I think we'll need #1434 to be finished to unblock CI in this repo.

jameslamb commented:
One the notebooks is failing with what looks like a new `cudf`-compatibility issue. Wrote that up here: https://github.com/rapidsai/cuspatial/issues/1437

jakirkham commented:
Going to covert to draft for clarity while those other blockers get cleared out

harrism commented:
I think draft should be reserved for PRs unready for review. This PR is already reviewed, but can't be merged. I suggest using the DO NOT MERGE label, which does what it says on the tin.

jameslamb commented:
> I think draft should be reserved for PRs unready for review.

No problem, I've moved this back to `Ready for Review` and added the `DO NOT MERGE` label.

jameslamb commented:
Hopefully CI here will pass once #1438 is merged.

jameslamb commented:
Thanks so much for all the help, everyone! Sorry this took a couple iterations, but I think this testing is in a better state now.
